course-instructions.md

```diff
@@ -0,0 +1,342 @@
+  - There are two types of targets - objects and files. If your target is a file, you need to add `format = "file"` to your `tar_target()` call and the command needs to return the filename of the new file.
+  - Setup the full pipeline by combining all targets into a single `list` object.
+  - There are 2 ways to define packages used to build targets: 1) declare using the `packages` argument in `tar_option_set()` in your makefile to specify packages used by all targets or 2) use the `packages` argument in individual `tar_target()` functions for packages that are specific to those targets.
+  - `model_RMSEs_csv` shows up two times - why? `model_RMSEs_csv` is the name of a target that creates the file `model_RMSEs.csv` when the command `download_data()` is run. When passed in as input to other functions (unquoted), it represents the filename of the file that was created when it was built. So when `model_RMSEs_csv` shows up as an _argument_ to another function, `process_data()`, it is really passing in the filename. The `process_data()` function then reads the files and changes the data (or "processes" it) in some way. 
+  
+---
+
+We're going to start with this simple example, and modify it to match our pipeline structure. This will start by creating a new branch, creating a new file, adding that file to git tracking, and opening a new pull request that includes the file:
+
+### :keyboard: Activity: get your code plugged into a makefile
+
+First things first: We're going to want a new branch. You can delete your previous one, since that pull request was merged. 
+```
+git checkout main
+git pull
+git branch -d structure
+git checkout -b makefile
+git push -u origin makefile 
+```
+---
+Next, create the file with the contents we've given you by entering the following from your repo directory in terminal/command line:
+```
+cat > _targets.R
+library(targets)
+source("code.R")
+tar_option_set(packages = c("tidyverse", "sbtools", "whisker"))
+
+list(
+  # Get the data from ScienceBase
+  tar_target(
+    model_RMSEs_csv,
+    download_data(out_filepath = "model_RMSEs.csv"),
+    format = "file"
+  ), 
+  # Prepare the data for plotting
+  tar_target(
+    eval_data,
+    process_data(in_filepath = model_RMSEs_csv),
+  ),
+  # Create a plot
+  tar_target(
+    figure_1_png,
+    make_plot(out_filepath = "figure_1.png", data = eval_data), 
+    format = "file"
+  ),
+  # Save the processed data
+  tar_target(
+    model_summary_results_csv,
+    write_csv(eval_data, file = "model_summary_results.csv"), 
+    format = "file"
+  ),
+  # Save the model diagnostics
+  tar_target(
+    model_diagnostic_text_txt,
+    generate_model_diagnostics(out_filepath = "model_diagnostic_text.txt", data = eval_data), 
+    format = "file"
+  )
+)
+```    
+
+then use `Ctrl+D` to exit the file creation mode and return to the prompt. 
```

Comment:
This is fine, but curious as to why this is done in terminal

Reply:
This is a hold over from the dynamic training and was probably part of the original scipiper training. I think it is probably a verrry gentle introduction to using `vim` to do stuff.